lib/client-assets.php

```diff
@@ -446,3 +446,43 @@ function gutenberg_register_packages_styles( $styles ) {
 	$styles->add_data( 'wp-widgets', 'rtl', 'replace' );
 }
 add_action( 'wp_default_styles', 'gutenberg_register_packages_styles' );
+
+/**
+ * Resolves WP Dependency handles to HTML.
+ *
+ * @param array      $instance WP Dependency instance.
+ * @param array|null $handles  Handles to resolve.
+ *
+ * @return string HTML.
+ */
+function gutenberg_resolve_dependencies( $instance, $handles ) {
+	if ( ! $handles || count( $handles ) === 0 ) {
+		return '';
+	}
+
+	ob_start();
+
+	$done           = $instance->done;
+	$instance->done = array();
+	$instance->do_items( array_unique( $handles ) );
+	$instance->done = $done;
+
+	return ob_get_clean();
+}
+
+add_filter(
+	'block_editor_settings_all',
+	function( $settings ) {
+		// These properties are for internal use only.
+		$settings['__unstableResolvedAssets'] = gutenberg_resolve_assets();
```

Comment:
This is already part of [this](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.0/client-assets.php#L109). Do we need it here as well?

It also reminds me that we may want to move this code to either `lib/compat/wordpress-6.1` (if we aim to ship it in the next WP) or `lib/experimental` (if it's not to be ported to core yet).

Reply:
Hm, seems like we don't need it here then. Thanks!
Ok, I'll move it to lib/experimental